sources/identify.php

```diff
@@ -2385,12 +2389,16 @@ function shouldUserAuthWithOauth2(
                 return [
                     'error' => false,
                     'message' => '',
+                    'oauth2Connection' => true,
+                    'userPasswordVerified' => true,
                 ];
             } elseif ((string) $userInfo['auth_type'] !== 'oauth2') {
                 // Case where auth_type is not managed
                 return [
                     'error' => true,
                     'message' => 'user_not_allowed_to_auth_to_teampass_app',
+                    'oauth2Connection' => false,
+                    'userPasswordVerified' => false,
                 ];
             }
```

Comment:
An OAuth2 user already known by teampass enters the last return: 
```php
        'oauth2Connection' => false,
        'userPasswordVerified' => false,
```

I suggest to modify this block as follows:
```php
            } elseif ((string) $userInfo['auth_type'] === 'oauth2') {
                // OAuth2 login request on OAuth2 user account.
                return [
                    'error' => false,
                    'message' => '',
                    'oauth2Connection' => true,
                    'userPasswordVerified' => true,
                ];
            } else {
                // Case where auth_type is not managed
                return [
                    'error' => true,
                    'message' => 'user_not_allowed_to_auth_to_teampass_app',
                    'oauth2Connection' => false,
                    'userPasswordVerified' => false,
                ];
            }
```